fedimint-server/src/config.rs

```diff
@@ -97,12 +96,13 @@ pub struct ServerConfigLocal {
     pub modules: BTreeMap<String, serde_json::Value>,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct SerdeCert(#[serde(with = "serde_tls_cert")] pub rustls::Certificate);
-
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Peer {
-    pub hbbft: ConnectionConfig,
+    /// Certs for TLS communication, required for peer authentication
+    #[serde(with = "serde_tls_cert")]
+    pub tls_cert: rustls::Certificate,
+    /// The TLS network address and port, used for HBBFT consensus
+    pub hbbft: SocketAddr,
```

Comment:
For now socket addresses are kind of ok (can work around it for now), but when connecting to some destination we generally want to be able to specify domains. You never want to hard-code IPs in production deployments since eventually you'll want to migrate and can't take the IP with you.

We might have to define our own protocol prefix (e.g. `fedimint://…` or an existing like `tls://…`) to use the `Url` type. 

To the question of how to make the fake network stack work: we can have our URLs be `fedimint://<ip>:<port>` in tests and just match the `<ip>:<port>` part with the bind address in the fake network stack.